File: python/swagger_client/models/record.py
# coding: utf-8

class Record(object):
    swagger_types = {
        'response': 'Response',
        'id': 'int',
        'reference': 'str',
        'business': 'Business',
        'contact': 'Contact',
        'po': 'str',
        'invoice': 'str',
        'customer': 'str',
        'tax_exempt': 'bool',
        'note': 'str',
        'credit_cards': 'list[VaultCreditCard]',
        'checks': 'list[VaultCheck]'
    }

    attribute_map = {
        'response': 'Response',
        'id': 'ID',
        'reference': 'Reference',
        'business': 'Business',
        'contact': 'Contact',
        'po': 'PO',
        'invoice': 'Invoice',
        'customer': 'Customer',
        'tax_exempt': 'TaxExempt',
        'note': 'Note',
        'credit_cards': 'CreditCards',
        'checks': 'Checks'
    }

    def __init__(self, response=None, id=None, reference=None, business=None, contact=None, po=None, invoice=None, customer=None, tax_exempt=None, note=None, credit_cards=None, checks=None):  # noqa: E501
        """Record - a model defined in Swagger"""  # noqa: E501
        self._response = None
        self._id = None
        self._reference = None
        self._business = None
        self._contact = None
        self._po = None
        self._invoice = None
        self._customer = None
        self._tax_exempt = None
        self._note = None
        self._credit_cards = None
        self._checks = None
        self.discriminator = None
        if response is not None:
            self.response = response
        if id is not None:
            self.id = id
        if reference is not None:
            self.reference = reference
        if business is not None:
            self.business = business
        if contact is not None:
            self.contact = contact
        if po is not None:
            self.po = po
        if invoice is not None:
            self.invoice = invoice
        if customer is not None:
            self.customer = customer
        if tax_exempt is not None:
            self.tax_exempt = tax_exempt
        if note is not None:
            self.note = note
        if credit_cards is not None:
            self.credit_cards = credit_cards
        if checks is not None:
            self.checks = checks

    @property
    def response(self):
        return self._response

    @response.setter
    def response(self, response):
        self._response = response

    @property
    def id(self):
        return self._id

    @id.setter
    def id(self, id):
        self._id = id

    @property
    def reference(self):
        return self._reference

    @reference.setter
    def reference(self, reference):
        self._reference = reference

    @property
    def business(self):
        return self._business

    @business.setter
    def business(self, business):
        self._business = business

    @property
    def contact(self):
        return self._contact

    @contact.setter
    def contact(self, contact):
        self._contact = contact

    @property
    def po(self):
        return self._po

    @po.setter
    def po(self, po):
        self._po = po

    @property
    def invoice(self):
        return self._invoice

    @invoice.setter
    def invoice(self, invoice):
        self._invoice = invoice

    @property
    def customer(self):
        return self._customer

    @customer.setter
    def customer(self, customer):
        self._customer = customer

    @property
    def tax_exempt(self):
        return self._tax_exempt

    @tax_exempt.setter
    def tax_exempt(self, tax_exempt):
        self._tax_exempt = tax_exempt

    @property
    def note(self):
        return self._note

    @note.setter
    def note(self, note):
        self._note = note

    @property
    def credit_cards(self):
        return self._credit_cards

    @credit_cards.setter
    def credit_cards(self, credit_cards):
        self._credit_cards = credit_cards

    @property
    def checks(self):
        return self._checks

    @checks.setter
    def checks(self, checks):
        self._checks = checks

File: php/SwaggerClient-php/lib/Model/Record.php
<?php

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

class Record implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    protected static $swaggerModelName = 'Record';

    protected static $swaggerTypes = [
        'response' => '\Swagger\Client\Model\Response',
        'id' => 'int',
        'reference' => 'string',
        'business' => '\Swagger\Client\Model\Business',
        'contact' => '\Swagger\Client\Model\Contact',
        'po' => 'string',
        'invoice' => 'string',
        'customer' => 'string',
        'tax_exempt' => 'bool',
        'note' => 'string',
        'credit_cards' => '\Swagger\Client\Model\VaultCreditCard[]',
        'checks' => '\Swagger\Client\Model\VaultCheck[]'
    ];

    protected static $swaggerFormats = [
        'response' => null,
        'id' => 'int32',
        'reference' => null,
        'business' => null,
        'contact' => null,
        'po' => null,
        'invoice' => null,
        'customer' => null,
        'tax_exempt' => null,
        'note' => null,
        'credit_cards' => null,
        'checks' => null
    ];

    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    protected static $attributeMap = [
        'response' => 'Response',
        'id' => 'ID',
        'reference' => 'Reference',
        'business' => 'Business',
        'contact' => 'Contact',
        'po' => 'PO',
        'invoice' => 'Invoice',
        'customer' => 'Customer',
        'tax_exempt' => 'TaxExempt',
        'note' => 'Note',
        'credit_cards' => 'CreditCards',
        'checks' => 'Checks'
    ];

    protected static $setters = [
        'response' => 'setResponse',
        'id' => 'setId',
        'reference' => 'setReference',
        'business' => 'setBusiness',
        'contact' => 'setContact',
        'po' => 'setPo',
        'invoice' => 'setInvoice',
        'customer' => 'setCustomer',
        'tax_exempt' => 'setTaxExempt',
        'note' => 'setNote',
        'credit_cards' => 'setCreditCards',
        'checks' => 'setChecks'
    ];

    protected static $getters = [
        'response' => 'getResponse',
        'id' => 'getId',
        'reference' => 'getReference',
        'business' => 'getBusiness',
        'contact' => 'getContact',
        'po' => 'getPo',
        'invoice' => 'getInvoice',
        'customer' => 'getCustomer',
        'tax_exempt' => 'getTaxExempt',
        'note' => 'getNote',
        'credit_cards' => 'getCreditCards',
        'checks' => 'getChecks'
    ];

    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    public static function setters()
    {
        return self::$setters;
    }

    public static function getters()
    {
        return self::$getters;
    }

    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    protected $container = [];

    public function __construct(array $data = null)
    {
        $this->container['response'] = isset($data['response']) ? $data['response'] : null;
        $this->container['id'] = isset($data['id']) ? $data['id'] : null;
        $this->container['reference'] = isset($data['reference']) ? $data['reference'] : null;
        $this->container['business'] = isset($data['business']) ? $data['business'] : null;
        $this->container['contact'] = isset($data['contact']) ? $data['contact'] : null;
        $this->container['po'] = isset($data['po']) ? $data['po'] : null;
        $this->container['invoice'] = isset($data['invoice']) ? $data['invoice'] : null;
        $this->container['customer'] = isset($data['customer']) ? $data['customer'] : null;
        $this->container['tax_exempt'] = isset($data['tax_exempt']) ? $data['tax_exempt'] : null;
        $this->container['note'] = isset($data['note']) ? $data['note'] : null;
        $this->container['credit_cards'] = isset($data['credit_cards']) ? $data['credit_cards'] : null;
        $this->container['checks'] = isset($data['checks']) ? $data['checks'] : null;
    }

    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    public function getResponse()
    {
        return $this->container['response'];
    }

    public function setResponse($response)
    {
        $this->container['response'] = $response;

        return $this;
    }

    public function getId()
    {
        return $this->container['id'];
    }

    public function setId($id)
    {
        $this->container['id'] = $id;

        return $this;
    }

    public function getReference()
    {
        return $this->container['reference'];
    }

    public function setReference($reference)
    {
        $this->container['reference'] = $reference;

        return $this;
    }

    public function getBusiness()
    {
        return $this->container['business'];
    }

    public function setBusiness($business)
    {
        $this->container['business'] = $business;

        return $this;
    }

    public function getContact()
    {
        return $this->container['contact'];
    }

    public function setContact($contact)
    {
        $this->container['contact'] = $contact;

        return $this;
    }

    public function getPo()
    {
        return $this->container['po'];
    }

    public function setPo($po)
    {
        $this->container['po'] = $po;

        return $this;
    }

    public function getInvoice()
    {
        return $this->container['invoice'];
    }

    public function setInvoice($invoice)
    {
        $this->container['invoice'] = $invoice;

        return $this;
    }

    public function getCustomer()
    {
        return $this->container['customer'];
    }

    public function setCustomer($customer)
    {
        $this->container['customer'] = $customer;

        return $this;
    }

    public function getTaxExempt()
    {
        return $this->container['tax_exempt'];
    }

    public function setTaxExempt($tax_exempt)
    {
        $this->container['tax_exempt'] = $tax_exempt;

        return $this;
    }

    public function getNote()
    {
        return $this->container['note'];
    }

    public function setNote($note)
    {
        $this->container['note'] = $note;

        return $this;
    }

    public function getCreditCards()
    {
        return $this->container['credit_cards'];
    }

    public function setCreditCards($credit_cards)
    {
        $this->container['credit_cards'] = $credit_cards;

        return $this;
    }

    public function getChecks()
    {
        return $this->container['checks'];
    }

    public function setChecks($checks)
    {
        $this->container['checks'] = $checks;

        return $this;
    }
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}
